packages/flutter/lib/src/widgets/focus_traversal.dart

```diff
@@ -1618,19 +1672,32 @@ class FocusTraversalGroup extends StatefulWidget {
   }
 }
 
+// A special focus node subclass that only FocusTraversalGroup uses so that it
+// can be used to cache the policy in the focus tree, and so that the traversal
+// code can find groups in the focus tree.
+class _FocusTraversalGroupNode extends FocusNode {
+  _FocusTraversalGroupNode({
+    super.debugLabel,
+    required this.policy,
+  });
+
+  FocusTraversalPolicy policy;
+}
+
 class _FocusTraversalGroupState extends State<FocusTraversalGroup> {
   // The internal focus node used to collect the children of this node into a
   // group, and to provide a context for the traversal algorithm to sort the
-  // group with.
-  late final FocusNode focusNode;
+  // group with. It's a special subclass of FocusNode just so that it can be
+  // identified when walking the focus tree during traversal, and hold the
+  // current policy.
+  late final _FocusTraversalGroupNode focusNode;
 
   @override
   void initState() {
     super.initState();
-    focusNode = FocusNode(
-      canRequestFocus: false,
-      skipTraversal: true,
+    focusNode = _FocusTraversalGroupNode(
       debugLabel: 'FocusTraversalGroup',
+      policy: widget.policy,
     );
   }
```

Comment:
I think this can be simplified to

```suggestion
  late final _FocusTraversalGroupNode focusNode = _FocusTraversalGroupNode(
    debugLabel: 'FocusTraversalGroup',
    policy: widget.policy,
  );
```